lib/scep/pki_operation/request.rb

```diff
@@ -57,6 +77,18 @@ def encrypt(target_encryption_certs)
         sign_and_encrypt_raw(csr.to_der, target_encryption_certs)
       end
 
+      # Encrypts with extended attributes required by some SCEP clients.
+      # @note THIS WILL BREAK VERIFICATION!
+      def encrypt_with_extended_attributes(target_encryption_certs)
+        raise ArgumentError, '#csr must be an OpenSSL::X509::Request' unless
+          csr.is_a?(OpenSSL::X509::Request)
+        pkcs7 = sign_and_encrypt_raw(csr.to_der, target_encryption_certs)
+        add_scep_message_type(pkcs7)
+      end
+
+      # TODO: uncomment/refactor once adding extended attributes works
+      # alias_method :encrypt, :encrypt_with_extended_attributes
```

Comment:
Uncomment this line to make this work with EJBCA. However, this will break verifying the newly encoded message with ruby (bypassable with `OpenSSL::PKCS7::NOVERIFY`)